crypto/s2n_ecc_evp.c

```diff
@@ -133,11 +136,6 @@ static int s2n_ecc_evp_compute_shared_secret(EVP_PKEY *own_key, EVP_PKEY *peer_p
     notnull_check(peer_public);
     notnull_check(own_key);
 
-    /* Peers MUST validate each other’s public key */
```

Comment:
Why don't they need to validate each other's pubkey any longer?

Reply:
We required to validate each other's public key still, I shall re-open this issue https://github.com/awslabs/s2n/issues/1404. 

The reason I had to revert this addition is because `EC_KEY_check_key` API that we used requires an ec_key object, and for Curve x25519 an ec_key object cannot be obtained from the evp_pkey and it returns a NULL value (which I didn't foresee for this case).  There is no alternative to this function in the `EVP_PKEY` interface for OpenSSL versions 1.1.0 and lower. The functions `EVP_PKEY_check` and `EVP_PKEY_public_check` are available only in OpenSSL version 1.1.1 and higher.

Reply:
Can you add peer public-key validation IFF OpenSSL version >= 1.1.1 to the test branch for now?

Reply:
I'm going to "Well, actually..." myself. After discussion with SalusaSecondus and reviewing https://tools.ietf.org/html/rfc8446#section-4.2.8.2, the peer public-key validation check doesn't apply to x25519. For more discussion, see https://moderncrypto.org/mail-archive/curves/2017/000896.html